### ISSUE TITLE
Introduce DDP

### ISSUE THREAD

**Issue by purzelrakete** [feldberlin/wavenet#17]
## What

Use nn.DistributedDataParallel instead of nn.DataParallel. 

## Why

The immediate motivation is to use SynchBatchNorm, which is only available for DDP. However DDP is also the recommended data parallel method in pytorch. As it turns out, the machinery which has to be added is pretty minimal when using fork process. 

## Acceptance Criteria

- [x] Implemented
- [x] Training works
- [x] DataLoader does the right thing
- [x] wandb does the right thing
- [x] Checkpointing does the right thing
- [x] Restoring checkpoints does the right thing
- [x] Make sure sync batchnorm is implemented and working

**Comment by purzelrakete** [comment]
```
bin/train maestro -p batch_size 12 -p batch_norm True -p learning_rate 0.05 -p max_epochs 12
```
